### PR TITLE
[prometheus] fix check metrics parsing

### LINKED ISSUE #12317: prometheus_scrape.checks fails to parse metrics list
**Describe what happened:**
According to documentation, Prometheus checks allow to specify metrics as:
- list of strings
- list of key-value pairs for renaming metrics
- list of key -> object, specifying name, type and other properties to further customize metrics

But when configuring checks with Helm using `prometheusScrape.additionalConfigs` (which translates to `DD_PROMETHEUS_SCRAPE_CHECKS` env variable in the agents) , the parsing fails and only allows a list of strings.

Example:
```yaml
      prometheusScrape:
        enabled: true
        additionalConfigs:
        - configurations:
            metrics:
            - foo: bar
           autodiscovery:
             kubernetes_container_names:
             - foo
```
Or:
`DD_PROMETHEUS_SCRAPE_CHECKS=[{"autodiscovery":{"kubernetes_container_names":["foo"]},"configurations":[{"metrics":[{"foo":"bar"}]}]}]`

In `agent config` we can see this is forced as a list of strings and ends up as an empty string:
```
  - configurations:
    - metrics:
      - ""
    autodiscovery:
      kubernetes_container_names:
      - foo
```

I think this is because Metrics si defined as a []string in [this struct](https://github.com/DataDog/datadog-agent/blob/main/pkg/autodiscovery/common/types/prometheus.go#L68).

From what I understand, there's no actual reason why we want to even parse Metrics, since this prometheus instances are later transformed to JSON again [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/autodiscovery/common/utils/prometheus.go#L52) for usage in the Agent.

If my understanding is correct, we could migrate Metrics to type `json.RawMessage` to just skip parsing this, and delegate this to the prometheus check.
Another approach would be to have multiple structs with different possible types and try to parse every possible case, but this is not ideal.

### PR DESCRIPTION
**Pull request by sschepens**
### What does this PR do?

Makes `PrometheusCheck` Metrics an array of `interface{}` so we don't force individual items to only be strings.

### Motivation

Fixes https://github.com/DataDog/datadog-agent/issues/12317

### Additional Notes

Added a test for the transformation of `PrometheusCheck` to an `Integration.Data`.

There seems to be no tests on the transformation of `DD_PROMETHEUS_SCRAPE_CHECKS` env variable to a prometheus check.

### Possible Drawbacks / Trade-offs

If tests pass, there should be no drawback to this change.

### Describe how to test/QA your changes

Tests included should be enough I guess.

_Edit from the reviewer:_
- _Define `metrics` as a map like the example mentioned in this issue https://github.com/DataDog/datadog-agent/issues/12317_
- _Define `metrics` as a list and make sure there are no regressions_

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by djmitche — APPROVED** [review]

**Review by ahmed-mez — COMMENTED** [review]
Hi @sschepens 

Thank you for the contribution! Could you just add a release note - the process is documented here https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno

**Comment by sschepens** [comment]
@ahmed-mez added release notes

**Review by ahmed-mez — APPROVED** [review]
Thank you!

File: releasenotes/notes/fix-prometheus-checks-metrics-07509580eb216c76.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fix prometheus check Metrics parsing, don't enforce a list of strings
```
Comment (jtappa) [inline releasenotes/notes/fix-prometheus-checks-metrics-07509580eb216c76.yaml:11]:
```suggestion
    Fix prometheus check Metrics parsing by not enforcing a list of strings.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Comment by sschepens** [comment]
@jtappa accepted suggestion

**Review by jtappa — APPROVED** [review]

**Comment by sschepens** [comment]
@ahmed-mez is it possible to include this fix in a patch? or do I have to wait for a full release?

**Comment by ahmed-mez** [comment]
@sschepens this fix will be available in agent v7.38 (to be released in ~2 months)

**Pull request merged**